File: workingtime/src/main/java/io/nirahtech/erp/workingtime/Imputation.java
package io.nirahtech.erp.workingtime;

import java.time.LocalDate;
import java.util.concurrent.TimeUnit;

public class Imputation {
    private final LocalDate date;
    private final int number;
    private final TimeUnit timeUnit;
    private final Project project;
    private final String details;

    public Imputation(LocalDate date, int number, TimeUnit timeUnit, Project project, String details) {
        this.date = date;
        this.number = number;
        this.timeUnit = timeUnit;
        this.project = project;
        this.details = details;
    }

    public LocalDate getDate() {
        return date;
    }

    public int getNumber() {
        return number;
    }

    public TimeUnit getTimeUnit() {
        return timeUnit;
    }

    public Project getProject() {
        return project;
    }

    public String getDetails() {
        return details;
    }
}

File: webapp/src/main/react/src/data/Imputation.ts
import Project from "./Project";
import TimeUnit from "./TimeUnit";

class Imputation {
    private readonly date: Date;
    private readonly number: number;
    private readonly timeUnit: TimeUnit;
    private readonly project: Project;
    private readonly details: string;

    constructor(date: Date, number: number, timeUnit: TimeUnit, project: Project, details: string) {
        this.date = date;
        this.number = number;
        this.timeUnit = timeUnit;
        this.project = project;
        this.details = details;
    }

    public getDate(): Date {
        return this.date;
    }

    public getNumber(): number {
        return this.number;
    }

    public getTimeUnit(): TimeUnit {
        return this.timeUnit;
    }

    public getProject(): Project {
        return this.project;
    }

    public getDetails(): string {
        return this.details;
    }
}

export default Imputation;